File: NUnit.vb.Test/Examples/TypeOfTest.vb
Option Explicit On

Imports NUnit.Framework

Namespace Examples

	<TestFixture>
	Public Class TypeOfTest

		<Test>
		Public Sub TypeOf_ListOfInt()

			Dim testList AS New List(Of Integer)
			Assert.That(testList, [Is].TypeOf(GetType(List(Of Integer))))

		End Sub

	End Class

End Namespace

File: Nunit.Test/Examples/TypeOfTest.cs
using NUnit.Framework;
using System.Collections.Generic;

namespace Nunit.Test.Examples
{
	[TestFixture]
	public class TypeOfTest
	{
		[Test]
		public void TypeOf_ListOfInt()
		{
			var theList = new List<int>();

			Assert.That(theList, Is.TypeOf<List<int>>());
		}
	}
}